topics/icrc_34_get_delegation.md

```diff
@@ -0,0 +1,192 @@
+<!-- TOC -->
+
+## Summary
+
+When a relying party wants to remove wallet approval prompts and sign on behalf of its users, it uses the
+below `icrc34_get_delegation` method to obtain a delegation chain.
+
+The delegation chain's identifier may the user's wallet address (Wallet Delegation) or an address meant to be exclusively used 
+for the relying party (App-Specific Delegation).
+
+Wallet Delegations **MUST** be restricted by the signer to canister targets that trust the relying party following the 
+[ICRC-28](./icrc_28_trusted_origins.md) standard. Doing otherwise would allow malicious actors to steal user assets.
+
+App-Specific Delegations **MUST** be restricted by the signer to individual relying parties.
+
+Signers **SHOULD** give users the choice to authenticate with their wallets (if available) or with an app-specific identifier.
+
+Relying parties **SHOULD** remove targets if they want to isolate users from the rest of the ecosystem.
+
+## Method
+
+**Name and Scope:** `icrc34_get_delegation`
+
+**Prerequisite:** ICRC-25 granted permission scope `icrc34_get_delegation` or `*`.
+
+## Scope (according to the [ICRC-25 standard](./icrc_25_signer_interaction_standard.md))
+
+**Scope:** `icrc34_get_delegation`
+
+**Required Properties:**
+
+- `publicKey` (`text` string): An Ed25519 or ECDSA session key as described [here](https://internetcomputer.org/docs/current/references/ic-interface-spec/#signatures).
+
+**Optional Properties:**
+
+- `targets` (`text` array): A list of target canister ids (textual representation) the scope is restricted to.
+- `maxTimeToLive` (`text` array): Expiration of the delegation in nanoseconds, though signer can still choose to
+return a delegation with a shorter expiration.
+- `idleOptions`
+
+### Example RPC Permission Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc25_request_permissions",
+  "params": {
+    "scopes": [
+      {
+        "method": "icrc34_get_delegation"
+      }
+    ]
+  }
+}
+```
+
+## Requests
+
+### Example RPC Get Delegation Request
+
+Requests for a delegation can optionally include `targets`, enabling signers to verify the security of returning a wallet 
+delegation (via ICRC-28). Signers in this case **SHOULD** offer users the option to connect to the application with either 
+a wallet delegation (limited to canisters listed in `targets`) or an app-specific delegation (without targets). If the array
+of `targets` is empty, signers **SHOULD** only offer users the option to connect with an app-specific delegation.
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrc34_get_delegation",
+  "params": {
+    "publicKey": "MDwwDAYKKwYBBAGDuEMBAgMsAAoAAAAAAGAAJwEB9YN/ErQ8yN+14qewhrU0Hm2rZZ77SrydLsSMRYHoNxM=",
+    "targets": [
+      "xhy27-fqaaa-aaaao-a2hlq-cai"
+    ],
+    "maxTimeToLive": "28800000000000"
+  }
+}
+```
+
+### Example RPC Get Delegation Response (wallet delegation)
+
+`publicKey`: Public key of delegation identity as described in
```

Comment:
This should be defined in a proper `Response` section rather than inline with the examples (which now is also duplicated).